django/db/backends/mysql/operations.py

```diff
@@ -182,25 +182,33 @@ def combine_expression(self, connector, sub_expressions):
     def get_db_converters(self, expression):
         converters = super(DatabaseOperations, self).get_db_converters(expression)
         internal_type = expression.output_field.get_internal_type()
-        if internal_type in ['BooleanField', 'NullBooleanField']:
-            converters.append(self.convert_booleanfield_value)
-        if internal_type == 'UUIDField':
-            converters.append(self.convert_uuidfield_value)
         if internal_type == 'TextField':
             converters.append(self.convert_textfield_value)
+        elif internal_type in ['BooleanField', 'NullBooleanField']:
+            converters.append(self.convert_booleanfield_value)
+        elif internal_type == 'DateTimeField':
+            converters.append(self.convert_datetimefield_value)
+        elif internal_type == 'UUIDField':
+            converters.append(self.convert_uuidfield_value)
         return converters
 
+    def convert_textfield_value(self, value, expression, connection, context):
+        if value is not None:
+            value = force_text(value)
+        return value
+
     def convert_booleanfield_value(self, value, expression, connection, context):
         if value in (0, 1):
             value = bool(value)
         return value
 
-    def convert_uuidfield_value(self, value, expression, connection, context):
+    def convert_datetimefield_value(self, value, expression, connection, context):
         if value is not None:
-            value = uuid.UUID(value)
+            if settings.USE_TZ:
```

Comment:
All other `convert_*_value` methods are written in the style I used here, i.e. `if value is not None: <do stuff>` . I normalized towards the most commonly used style to minimize code churn.

I don't have a particular preference for this style vs. what you're suggesting, I would simply like to keep the implementation style consistent.